src/Symfony/Bundle/FrameworkBundle/Controller/TemplateController.php

```diff
@@ -25,12 +24,10 @@
 class TemplateController
 {
     private $twig;
-    private $templating;
 
-    public function __construct(Environment $twig = null, EngineInterface $templating = null)
+    public function __construct(Environment $twig = null)
```

Comment:
I feel like twig should not be optional in this class. And when twig is not installed, the TemplateController should be removed from DI.

Reply:
I agree, but for DX point of view it's better as is, because you'll get a good error message:
> You can not use the TemplateController if the Twig Bundle is not available.

vs

> Failed asserting that exception of type "ArgumentCountError" matches expected exception "\LogicException". Message was: "Too few arguments to function Symfony\Bundle\FrameworkBundle\Controller\TemplateController::__construct(), 0 passed in ...

WDYT?

Reply:
I agree it's a little better for DX. But IMO it's not the responsiblity of the Controller do provide that.
We could just catch the instantiation error in https://github.com/symfony/symfony/blob/0119d217cad9cac6b2b6461c62fd879b64f16406/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php#L54 and throw a better exception. It's even in the same namespace and the correct class to enhance DX. Then it's clear it's just for DX and we don't need to hack around it in the actual implementations.